lib/Interpreter/IncrementalParser.cpp

```diff
@@ -280,6 +280,55 @@ namespace cling {
     return m_Consumer->getTransaction();
   }
 
+  const Transaction*
+  IncrementalParser::mergeTransactionsAfter(const Transaction *T, bool Prev) {
+    llvm::SmallVector<Transaction*, 8> merge;
+    for (std::deque<Transaction*>::reverse_iterator itr = m_Transactions.rbegin(),
+            end  = m_Transactions.rend(); itr != end; ++itr) {
+      Transaction *cur = *itr;
+      merge.push_back(cur);
+      if (cur==T) {
+        if (!Prev)
+          merge.pop_back();
+        break;
+      }
+    }
```

Comment:
If `T` is not a top-most transaction, or unloaded, or rolled back, then it will not be part of `m_Transactions`. In that case, your code currently merges *all* transactions into the first (if I read your code correctly).

You should at least assert that you've seen `T` in `m_Transactions`. I'd even simply use `find()` instead of building the auxiliary `SmallVector merge`.